src/mem.c

```diff
@@ -104,6 +104,24 @@ void ofi_mem_fini(void)
 	free(page_sizes);
 }
 
+size_t ofi_get_mem_size(void)
+{
+	long page_cnt, page_size;
+	size_t mem_size;
+
+	page_cnt = sysconf(_SC_PHYS_PAGES);
+	page_size = ofi_get_page_size();
+
+	if (page_cnt <= 0 || page_size <= 0)
+		return 0;
+
+	mem_size = (size_t) page_cnt * (size_t) page_size;
+	if (mem_size < page_cnt || mem_size < page_size)
```

Comment:
How can mem_size be less than page_cnt?

Reply:
According to the man pages, page_cnt * page_size can overflow.